### ISSUE TITLE
'Value' is not a subtype of type 'Map<String, dynamic>'

### ISSUE THREAD

**Issue by kaciula** [stvndelucis/firestore_api_parser#1]
Hi.

Thanks for the package. It's what I really needed. However, I can't seem to make it parse any Firestore document. Here is what I am doing.

  ```
Map<String, dynamic> trainerJson = firestoreParser.parseFirestoreDocument(
      documentJson: response.documents![0].toJson());
```
I get an error saying 'Value' is not a subtype of type 'Map<String, dynamic>'

I'm attaching a screenshot of how the document looks like before I attempt to parse it with your package. Is there a type of field that is not supported? Or what might be the issue?

![Screenshot 2022-02-10 at 12 33 22](https://user-images.githubusercontent.com/18688/153389323-fab3e82c-5a3b-46a3-8a15-b4b71ae9390d.png)

**Comment by stvndelucis** [comment]
Hi! I have received your message, I will examine your problem and see what
is wrong.

Thanks for using my package

Le jeu. 10 févr. 2022 à 11:35, Catalin Morosan ***@***.***> a
écrit :

**Comment by stvndelucis** [comment]
Can you attach me the .json file of your firestore json ? thanks

Le jeu. 10 févr. 2022 à 12:08, Steeven Aime ***@***.***> a
écrit :

**Comment by kaciula** [comment]
@stvndelucis Hi. Sorry, I haven't seen you latest message.

I've figured out the issue. I am using https://pub.dev/packages/googleapis to call the Firestore API. And when I call `.toJson()` on the returned documents, it does not map the internal fields to `Map<String, dynamic>` and that is the reason the `parseFirestoreDocument()` method fails.

I made a local copy of your package and calling `entry.value.toJson()` in several places. And I also parse the int/double values because the googleapis package returns the fields as `Strings`.

Not sure if there is anything to do on your part. Your package expects a valid json as a Map<dynamic, dynamic> and I wasn't providing exactly that because the googleapis packages isn't giving me that.
